syntax = "proto3";

option csharp_namespace = "GrpcNodeServer";


message UploadFileRequest {
	string file_id = 1;
	string type = 2;
	bytes file_content = 3;
	int32 user_id = 4;
}

message UploadFileResponse {
	bool status = 1;
	string message = 2;
}

message UpdateFileRequest {
	string file_id = 1;
	bytes new_content = 3;
	int32 user_id = 4;
}

message UpdateFileResponse {
	bool status = 1;
	string message = 2;
}

message DownloadFileRequest {
	string file_id = 1;
	int32 user_id = 4;
}

message DownloadFileResponse {
	bool status = 1;
	bytes file_content = 2;
	optional string message = 3;
}

message DeleteFileRequest {
	string file_id = 1;
	int32 user_id = 4;
}

message DeleteFileResponse {
	bool status = 1;
	string message = 2;
}

/*
message StartNodeGroupProposal
{
	
}
*/

service NodeServices {
  rpc UploadFile(stream UploadFileRequest) returns (UploadFileResponse);
  rpc UpdateFile(stream UpdateFileRequest) returns (UpdateFileResponse);
  rpc DownloadFile(DownloadFileRequest) returns (stream DownloadFileResponse);
  rpc DeleteFile(DeleteFileRequest) returns (DeleteFileResponse);
}